### PR TITLE
[Merged by Bors] - Changed code to adhere to the behavior defined in ADR18.

### PR DESCRIPTION
**Pull request by soenkeliebau**
# Description

This contains a breaking change, since the version that is specified for HDFS now has to include a stackable image version.

Previously the version could be specified as "version: 3.2.1" and the operator simply added the stackable version to this. But going forward we have decided that the image version has to be explictly specified, so it will now have to be "version: 3.2.1-stackable-0.1.0" or similar.

This is in accordance with ADR18 (https://docs.stackable.tech/home/contributor/adr/ADR018-product_image_versioning.html).

Also moved integration tests back into this repo.

*Please add a description here. This will become the commit message of the merge request later.*

## Review Checklist

- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: tests/test-definition.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+dimensions:
+  - name: druid
+    values:
+      - 0.22.1-stackable0.1.0
```
Comment (adwk67) [inline tests/test-definition.yaml:5]: `0.22.1-stackable0` (but not `0.1.0`) resolved to the image including the authorizer, so we need the `0.2.0` here :
```suggestion
      - 0.22.1-authorizer0.1.0-stackable0.2.0
```

**Review by adwk67 — CHANGES_REQUESTED** [review]

File: examples/psql-s3/psql-s3-druid-cluster.yaml [hunk 2]
```diff
@@ -35,7 +35,7 @@ kind: DruidCluster
 metadata:
   name: psql-s3-druid
 spec:
-  version: 0.22.1
+  version: 0.22.1-stackable0.1.0
```
Comment (adwk67) [inline examples/psql-s3/psql-s3-druid-cluster.yaml:38]:
```suggestion
  version: 0.22.1-authorizer0.1.0-stackable0.2.0
```

File: examples/psql/psql-hdfs-druid-cluster.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ kind: DruidCluster
 metadata:
   name: psql-druid
 spec:
-  version: 0.22.1
+  version: 0.22.1-stackable0.1.0
```
Comment (adwk67) [inline examples/psql/psql-hdfs-druid-cluster.yaml:30]:
```suggestion
  version: 0.22.1-authorizer0.1.0-stackable0.2.0
```

File: examples/simple-druid-cluster.yaml [hunk 4]
```diff
@@ -47,7 +47,7 @@ kind: DruidCluster
 metadata:
   name: simple-derby-druid
 spec:
-  version: 0.22.1
+  version: 0.22.1-stackable0.1.0
```
Comment (adwk67) [inline examples/simple-druid-cluster.yaml:50]:
```suggestion
  version: 0.22.1-authorizer0.1.0-stackable0.2.0
```

File: docs/modules/ROOT/pages/usage.adoc [hunk 5]
```diff
@@ -53,7 +53,7 @@ kind: DruidCluster
 metadata:
   name: simple-druid
 spec:
-  version: 0.22.1
+  version: 0.22.1-stackable0.1.0
```
Comment (adwk67) [inline docs/modules/ROOT/pages/usage.adoc:56]:
```suggestion
  version: 0.22.1-authorizer0.1.0-stackable0.2.0
```

File: tests/test-definition.yaml (hunk 1 above)
Comment (teozkr) [reply tests/test-definition.yaml:5]: We should probably fix this in the docker-images repo instead.

File: tests/test-definition.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+---
+dimensions:
+  - name: druid
+    values:
+      - 0.22.1-authorizer0.1.0-stackable0.1.0
```
Comment (adwk67) [inline tests/test-definition.yaml:5]:
```suggestion
      - 0.22.1-authorizer0.1.0-stackable0.2.0
```

**Review by adwk67 — APPROVED** [review]
lgtm.

File: tests/test-definition.yaml (hunk 1 above)
Comment (adwk67) [reply tests/test-definition.yaml]: Agreed - I'll open a separate issue for this.

**Pull request closed without merging**